src/ccAnnotateLensProvider.ts

```diff
@@ -2,26 +2,34 @@
 
 import * as vscode from 'vscode';
 import {ccAnnotateLens} from './ccAnnotateLens';
+import {ClearCase} from './clearcase'
 
 export class ccCodeLensProvider implements vscode.CodeLensProvider
 {
 	static selector = {
 		scheme: "file"
 	};
 
-	private m_context:vscode.ExtensionContext;
-
-	public constructor(context: vscode.ExtensionContext)
+	public constructor(private m_context: vscode.ExtensionContext, private m_clearcase: ClearCase)
 	{
-		this.m_context = context;
 	}
 
 	async provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.CodeLens[]>
 	{
 		let l_lenses: vscode.CodeLens[] = [];
-
-		l_lenses.push(new ccAnnotateLens(document, new vscode.Range(0,0,0,1)));
-
+		let l_isCcO: boolean;
+		try
+		{
+			l_isCcO = await this.m_clearcase.isClearcaseObject(document.uri);
+		}
+		catch(error)
+		{
+			l_isCcO = error;

```

Comment:
In this case my promise returns a bool and in case of a reject, error == false.
I needed the try catch to handle a promise reject. Before I done that, I got an error, not handling the reject.